exo-go/test_helpers/run_feature_context.go

```diff
@@ -0,0 +1,65 @@
+package testHelpers
+
+import (
+	"fmt"
+	"os"
+	"path"
+
+	"github.com/DATA-DOG/godog"
+	"github.com/DATA-DOG/godog/gherkin"
+	"github.com/Originate/exosphere/exo-go/src/docker_helpers"
+	"github.com/Originate/exosphere/exo-go/src/util"
+	"github.com/docker/docker/client"
+	"github.com/pkg/errors"
+)
+
+// RunFeatureContext defines the festure context for features/run
+// nolint gocyclo
+func RunFeatureContext(s *godog.Suite) {
+	var dockerClient *client.Client
+	var cwd, appName string
+
+	s.BeforeSuite(func() {
+		var err error
+		cwd, err = os.Getwd()
+		if err != nil {
+			panic(err)
+		}
+		dockerClient, err = client.NewEnvClient()
+		if err != nil {
+			panic(err)
+		}
+	})
+
+	s.Step(`^I am in the root directory of the "([^"]*)" example application$`, func(name string) error {
+		appDir = path.Join(cwd, "tmp", name)
+		appName = name
+		return checkoutApp(cwd, appName)
+	})
+
+	s.Step(`^my application has been set up correctly$`, func() error {
+		if err := setupApp(cwd, appName); err != nil {
+			return err
+		}
+		return nil
+	})
+
+	s.Step(`^my machine is running the services:$`, func(table *gherkin.DataTable) error {
+		var err error
+		dockerHelpers.ListRunningContainers(dockerClient, func(runningContainers []string, err error) {
```

Comment:
using a callback here because I couldn't figure a way to make this step pass without using the callback (see #331)